src/cpp/session/modules/data/SessionData.cpp

```diff
@@ -25,22 +27,168 @@ using namespace rstudio::core ;
 
 namespace rstudio {
 namespace session {
-namespace modules { 
+namespace modules {
 namespace data {
 
+class AsyncDataPreviewRProcess : public async_r::AsyncRProcess
+{
+public:
+   static boost::shared_ptr<AsyncDataPreviewRProcess> create(
+           const json::JsonRpcRequest& request,
+           const json::JsonRpcFunctionContinuation& continuation)
+   {
+      boost::shared_ptr<AsyncDataPreviewRProcess> pDataPreview(
+                  new AsyncDataPreviewRProcess(request, continuation));
+      pDataPreview->start();
+      return pDataPreview;
+   }
+
+   void terminateProcess()
+   {
+      async_r::AsyncRProcess::terminate();
+   }
+
+private:
+   AsyncDataPreviewRProcess(
+           const json::JsonRpcRequest& request,
+           const json::JsonRpcFunctionContinuation& continuation) :
+       continuation_(continuation),
+       request_(request)
+   {
+   }
+
+   FilePath pathFromModulesSource(std::string sourceFile)
+   {
+      FilePath modulesPath = session::options().modulesRSourcePath();
+      FilePath srcPath = modulesPath.complete(sourceFile);
+
+      return srcPath;
+   }
+
+   FilePath pathFromSource(std::string sourceFile)
+   {
+      FilePath sourcesPath = session::options().coreRSourcePath();
+      FilePath srcPath = sourcesPath.complete(sourceFile);
+
+      return srcPath;
+   }
+
+   void start()
+   {
+      std::ostringstream jsonStream;
+      json::Value requestValue = request_.params;
+      json::write(requestValue, jsonStream);
+      std::string jsonData = jsonStream.str();
+
+      std::string cmd = ".rs.preview_data_import_async('" + jsonData + "')";
+
+      std::vector<core::FilePath> sources;
+      sources.push_back(pathFromSource("Tools.R"));
+      sources.push_back(pathFromModulesSource("ModuleTools.R"));
+      sources.push_back(pathFromModulesSource("SessionDataViewer.R"));
+      sources.push_back(pathFromModulesSource("SessionDataImportV2.R"));
```

Comment:
In doing this just be careful that none of the methods you call attempt to call "rs_" functions (the ones embedded in the RStudio process) since they won't be available. You'll get an obvious error message if this occurs so your existing code is likely fine, just something to be aware of.